base/src/main/java/uk/ac/ebi/atlas/experimentimport/ExperimentDAO.java

```diff
@@ -104,59 +95,36 @@ private void addExperimentSpeciesRows(ExperimentDTO experimentDTO) {
     public void deleteExperiment(String experimentAccession) {
 
         int deletedRecordsCount = jdbcTemplate.update(DELETE_EXPERIMENT, experimentAccession);
-
-        if (deletedRecordsCount != 1) {
-            throw new IllegalArgumentException("Experiment not found for accession " + experimentAccession);
-        }
-    }
-
-    public ExperimentDTO findPublicExperiment(String experimentAccession) {
-        return findExperiment(experimentAccession, false);
+        checkState(deletedRecordsCount == 1 , "Experiment not found for accession " + experimentAccession);
```

Comment:
I would favour `checkArgument`: keeps the same exception semantics, `IllegalStateException` doesn’t make much sense here.